src/mon/PaxosService.cc

```diff
@@ -100,22 +100,15 @@ bool PaxosService::dispatch(MonOpRequestRef op)
 	   * Callback class used to propose the pending value once the proposal_timer
 	   * fires up.
 	   */
-	  class C_Propose : public Context {
-	    PaxosService *ps;
-	  public:
-	    explicit C_Propose(PaxosService *p) : ps(p) { }
-	    void finish(int r) override {
-	      ps->proposal_timer = 0;
+	  proposal_timer = new C_MonContext(mon, [this](int r) {
+	      proposal_timer = 0;
```

Comment:
This won't run in is_shutdown().  I think it's okay in this case since timer tolerates a cancel_event on a non-event, though, and who cares during shutdown.